src/us/kbase/test/auth2/service/ui/LoginTest.java

```diff
@@ -195,42 +195,46 @@ public void loginStartMinimalInput() throws Exception {
 		final NewCookie expectedredirect = new NewCookie("loginredirect", "no redirect",
 				"/login", null, "redirect url", 0, false);
 
-		loginStart(form, expectedsession, expectedredirect);
+		loginStart(form, expectedsession, expectedredirect, null);
 	}
 	
 	@Test
-	public void loginStartEmptyStrings() throws Exception {
+	public void loginStartEmptyStringsWithEnvironment() throws Exception {
 		final Form form = new Form();
 		form.param("provider", "prov1");
 		form.param("redirecturl", "  \t   \n   ");
 		form.param("stayloggedin", "  \t   \n   ");
+		form.param("environment", "myenv");
 		final NewCookie expectedsession = new NewCookie("issessiontoken", "true",
 				"/login", null, "session choice", 30 * 60, false);
 		final NewCookie expectedredirect = new NewCookie("loginredirect", "no redirect",
 				"/login", null, "redirect url", 0, false);
 
-		loginStart(form, expectedsession, expectedredirect);
+		loginStart(form, expectedsession, expectedredirect, "myenv");
 	}
 	
 	@Test
-	public void loginStartWithRedirectAndNonSessionCookie() throws Exception {
+	public void loginStartWithRedirectAndNonSessionCookieWithWhitespaceEnvironment()
+			throws Exception {
 		final String redirect = "https://foobar.com/thingy/stuff";
 		final Form form = new Form();
 		form.param("provider", "prov1");
 		form.param("redirecturl", redirect);
 		form.param("stayloggedin", "f");
+		form.param("environment", "   \t  ");
 		final NewCookie expectedsession = new NewCookie("issessiontoken", "false",
 				"/login", null, "session choice", 30 * 60, false);
 		final NewCookie expectedredirect = new NewCookie("loginredirect", redirect,
 				"/login", null, "redirect url", 30 * 60, false);
 
-		loginStart(form, expectedsession, expectedredirect);
+		loginStart(form, expectedsession, expectedredirect, null);
 	}
 
 	private void loginStart(
 			final Form form,
 			final NewCookie expectedsession,
-			final NewCookie expectedredirect)
+			final NewCookie expectedredirect,
+			final String environment)
```

Comment:
Just a question, how do you feel about trailing commas? They would make these diffs a little cleaner

Reply:
https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8

Reply:
That's a syntax error in Java unfortunately (or fortunately, depending on how you feel about unnecessary commas)